File: src/DeployManager.Client/src/shared/types/scheduledDeploy.ts
export interface ScheduledDeploy {
  id: string
  name: string
  siteId: string
  siteName: string
  packageId: string | null
  packageFileName: string | null
  scheduledAt: string
  status: string
  createdByUserName: string
  recipients: string[]
  notifyOnStart: boolean
  notifyOnComplete: boolean
  createdAt: string
  startedAt: string | null
  completedAt: string | null
  jobId: string | null
  errorMessage: string | null
}

export type ScheduledDeployDetail = ScheduledDeploy

export interface CreateScheduledDeployRequest {
  name: string
  siteId: string
  packageId: string | null
  scheduledAt: string
  recipients: string[]
  notifyOnStart: boolean
  notifyOnComplete: boolean
}

export interface UpdateScheduledDeployRequest {
  name: string
  scheduledAt: string
  recipients: string[]
  notifyOnStart: boolean
  notifyOnComplete: boolean
}

File: src/DeployManager.Domain/Entities/ScheduledDeploy.cs
using DeployManager.Domain.Common;
using DeployManager.Domain.Enums;

namespace DeployManager.Domain.Entities;

public class ScheduledDeploy : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public Guid SiteId { get; set; }
    public Guid? PackageId { get; set; }
    public DateTime ScheduledAt { get; set; }
    public ScheduledDeployStatus Status { get; set; } = ScheduledDeployStatus.Pending;
    public Guid CreatedByUserId { get; set; }
    public DateTime? StartedAt { get; set; }
    public DateTime? CompletedAt { get; set; }
    public Guid? JobId { get; set; }
    public string? ErrorMessage { get; set; }
    public string Recipients { get; set; } = "[]";
    public bool NotifyOnStart { get; set; } = true;
    public bool NotifyOnComplete { get; set; } = true;

    public DeploySite? Site { get; set; }
    public DeployPackage? Package { get; set; }
    public User? CreatedByUser { get; set; }
    public DeployJob? Job { get; set; }
}
